### PR TITLE
Support loading multiple comma-separated soundsfonts 

### PR DESCRIPTION
**Pull request by kcgen**
Multiple soundfonts can be loaded using a comma separator.

The existing singular format is supported:
 - `soundfont = name`
 - `soundfont = name scale`

No-whitespace with comma is supported:
 - `soundfont = name1,name2`
 - `soundfont = name1,name2 scale`

Whitespace with comma is supported:
 - `soundfont = name1, name2`
 - `soundfont = name1, name2 scale`

It gracefully handles missing soundfonts:
 - report an error and skips the missing one
 - It will still try to load those that it does find, in the order provided
 - If none are found, or none are loaded, then existing "full failure" of MIDI occurs

`mixer /listmidi` is supported and the conf documentation is updated.

### CODE REVIEW

**Comment by kcgen** [comment]
Just a draft for now: @GranMinigun will assess if this is worth-while or not.

**Comment by dreamer** [comment]
I asked about it several times and never got a proper answer: *why do we need this*?

In my opinion, this is a footgun that we're putting in front of the users for no apparent reason. I am opposed to this change until I'll know the rationale.

**Comment by kcgen** [comment]
See: https://www.vogons.org/viewtopic.php?t=35811

``` text
Legend:
NR = sends no reset
GM = sends GM1 reset
GS = sends GS reset

PERC = uses additional percussion notes (Roland GS)
RVB = cc91 reverb controller (Roland GS)
CRS = cc93 chorus controller (Roland GS)

BANK = uses extra instrument banks (Roland GS)
DRUM = uses special drum kits (Roland GS)
CTF = capital tone fallback

NRPN = special Roland controller (Roland GS)
SYSEX = special Roland system exclusive (Roland GS)
FLKEY = uses Roland SC-55 non-GM - Fl. Key Click (Roland GS)
++ = misc. enhancement, corrections
```

> I asked about it several times and never got a proper answer: _why do we need this_?

If the above allow games to go beyond the GM set and use GS calls (reverb, sticks, etc..), then multi-load will let users augment their GM soundfonts with GS-delta soundfonts, such as Fluid's GS delta-instrument.

I fully agree @dreamer - we definitely don't want this unless DOS games will actually take advantage of it.  Otherwise it's 100% unecessary complexity and over-engineering without a use-case.

PR is just a draft to let us explore this.

**Comment by dreamer** [comment]
> See: https://www.vogons.org/viewtopic.php?t=35811
> 
> ```
> Legend:
> NR = sends no reset
> GM = sends GM1 reset
> GS = sends GS reset
> 
> PERC = uses additional percussion notes (Roland GS)
> RVB = cc91 reverb controller (Roland GS)
> CRS = cc93 chorus controller (Roland GS)
> 
> BANK = uses extra instrument banks (Roland GS)
> DRUM = uses special drum kits (Roland GS)
> CTF = capital tone fallback
> 
> NRPN = special Roland controller (Roland GS)
> SYSEX = special Roland system exclusive (Roland GS)
> FLKEY = uses Roland SC-55 non-GM - Fl. Key Click (Roland GS)
> ++ = misc. enhancement, corrections
> ```

Quote from that thread:

> These hacks improve playback on modern MIDI hardware. Fixes random issues like:
> - device resets
> - wrong controller values
> - bad instruments
> - faulty drivers
> - limited channel range
> - unused features

As I understand it, DOS games running with GS experienced a number of bugs and issues (because they were not designed for it), and this topic lists hacks and workarounds to un-break the games and use GS despite the problems. Correct me if I understood the thread incorrectly.

> This PR is a draft to let us test this.

:+1:

**Comment by rderooy** [comment]
FYI, what I do is use this merged FluidR3 GM+GS soundfont https://musical-artifacts.com/artifacts/1229

**Comment by Wengier** [comment]
@rderooy Using a merged FluidR3 GM+GS soundfont is definitely a nice workaround. But I guess the solution provided in this PR is more flexible in general.

**Comment by HW-Scanno** [comment]
Hi, for the people that were lately helping me in discord giving me feedback about a soundfont creation, i wanted to join the discussion and write my 2 cents.
Being GS and XG are a superset of GM which means that other from the GM Instruments starting at bank 000:000 going up to 000:127 (Giving a total of 128 instruments) there are also Drum Kits, that holds a different drum instrument for each key pressed and those GM are at bank 128:000 going thru 128:127. Many of them are repeated meaning that only 10 different drumkits at maximum are used in GM.
GS then uses also the 127:000 to 127:127 bank giving alot more extra drumkits.
These are to be considered the difference in instruments.
There are also more differences in a GS midi file, for example the fact that you can store different version of an instrument, being said that now starts the problems.
There are no standards about GS or XG that tells me how many variations i have to include and which one.
Opposed to the GM standard which tells me which bank will always be a guitar and which bank will always be a violin, the GS standard doesn't tell me how many variations i need to include and which one should be...
Assuming that you can go blind, and let's say, you have to include a total of 4 different sounding guitars, who's gonna telling me that the Ibanez sampled guitar (for example) would go in the variation 1, 2 or 3 bank?
When you are left undetermined in your choices that put me in a situation to believe that there is no standard...
So for me apart the additional drumkits that is very well known in which position they are, GS standard is to not be considered.
Unless you have a specific case, or game, which needs you this or that instrument variation in a precise bank position, it's impossible to make one for all GS cases...
Also, adding a soundfont the additional GS drumkits is a matter of seconds with the Polyphone free software... Just highlight all the banks in 127 of a known GS font, and literally paste them in the GM font.
You will have your GM+GS font in a matter of seconds.

TLDR: GM is a very known standard with a list of instruments and their position, GS is a superset of GM with standard additional drumkits but undetermined in the position of additional instruments variations that are left to end users.
Being only a matter of seconds to add GS drumkits from an existing soundfont to another existing GM drumkit thru Polyphone it's just quickier to fix your soundfont by your hands than to implement that dubious stack of sf2 loaded one after another adding probably a significant overhead in playback...

**Comment by rderooy** [comment]
I think, from a DOS legacy gaming perspective, only the Roland SC55 needs to be considered for GS. So as long as you have the same number of variations, and they sound similar to a SC55, you should not have any issues.

**Comment by kcgen** [comment]
Thanks for the testing and discussion everyone!
Triage complete; closing.
